backend/src/utils/courses.js

```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    validateCourseId: (courseId) => {
```

Comment:
Isn't the course id stored as a number in the database? I suppose it would be meaningful to check that the courseId is a valid number instead of checking it is a number. Now it is a string just because it is read from the params and they are strings. Thus it would need to be converted as number when read with something like `const courseId = +req.params.id`. Then you could check it is not a NaN.